updater/config/config.go

```diff
@@ -0,0 +1,138 @@
+package config
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/url"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/onlineconf/onlineconf/updater/v3/updater"
+	"github.com/rs/zerolog/log"
+	"gopkg.in/yaml.v3"
+)
+
+type AdminConfig struct {
+	Endpoint string
+	// Deprecated: use Endpoint instead
+	Host string
+	// Deprecated: use Endpoint instead
+	Port     int
+	Username string
+	Password string
+	MTLS     *struct {
+		Cert string
+		Key  string
+	}
+}
+
+type ConfigFile struct {
+	Hostname       string
+	Datacenter     string
+	Admin          AdminConfig
+	DataDir        string `yaml:"data_dir"`
+	UpdateInterval int    `yaml:"update_interval"`
+	Variables      map[string]string
+}
+
+func formURI(endpoint string) (uri string, err error) {
+	adminURL, err := url.Parse(endpoint)
+	if err != nil {
+		err = fmt.Errorf("url.Parse: %w", err)
+		return
+	}
+	if adminURL.Scheme == "" || adminURL.Host == "" {
+		err = fmt.Errorf("endpoint requires scheme and host")
+		return
+	}
+	if len(adminURL.Path) > 0 && adminURL.Path[len(adminURL.Path)-1] == '/' {
+		adminURL.Path = adminURL.Path[:len(adminURL.Path)-1]
+	}
+	uri = adminURL.String()
+	return
+}
+
+func deprecatedFormURI(host string, port int) (uri string) {
+	switch port {
+	case 80:
+		uri = "http://" + host
+	case 443, 0:
+		uri = "https://" + host
+	default:
+		uri = "http://" + host + ":" + strconv.Itoa(port)
+	}
+	return uri
+}
+
+func parseMTLSConfig(cfg AdminConfig) (*tls.Config, error) {
+	if cfg.MTLS == nil {
+		return nil, nil
+	}
+
+	tlsCfg := *cfg.MTLS
+
+	certPEMBlock, err := os.ReadFile(tlsCfg.Cert)
```

Comment:
What do you think about embedding certificates into the yaml file (similar to kubectl, but using yaml's ability to embed multiline text (|) instead of base64)?
In any case, if you read from files, it's better to use tls.LoadX509KeyPair.

Reply:
Can't say I like the idea with embedded certificates ... It just seems not very user friendly

Reply:
I did add LoadX509KeyPair use